pkg/discovery/service.go

```diff
@@ -144,8 +160,39 @@ func (discovery *Discovery) refresh(ctx context.Context) error {
 	return nil
 }
 
+// graveyard attempts to ping all nodes in the Seen() map from Kademlia and adds them to the cache
+// if they respond. This is an attempt to resurrect nodes that may have gone offline in the last hour
+// and were removed from the cache due to an unsuccessful response.
+func (discovery *Discovery) graveyard(ctx context.Context) error {
+	discovery.log.Debug("starting node graveyard resurrection")
+	seen := discovery.kad.Seen()
+	var errors utils.ErrorGroup
+
+	for _, n := range seen {
+		ping, err := discovery.kad.Ping(ctx, *n)
+		if err != nil {
+			discovery.log.Debug("could not ping node in graveyard check")
```

Comment:
maybe add the nodeID to the log?

Reply:
I've been discouraged from adding node ID's to logs, so I didn't add it here.